src/Common/src/Interop/Windows/Kernel32/Interop.QueryDosDevice.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal partial class Interop
+{
+    internal partial class Kernel32
+    {
+          [DllImport(Libraries.Kernel32, SetLastError = true)]
+          internal static extern uint QueryDosDevice(string lpDeviceName, IntPtr lpTargetPath, int ucchMax);
```

Comment:
This should be `CharSet.Unicode` and ideally `QueryDosDeviceW` so that you can also set `ExactSpelling` to true in the `[DllImport]`.

Reply:
Ive added 

> ExactSpelling

Not sure about unicode? Would comm device names contain Unicode chars?
But guess QueryDosDeviceW exist for a reason?

Reply:
It is a historical thing that .NET picks the "A" version and uses ANSI by default. It is almost never the right choice as .NET and Windows use the same Unicode encoding.

Windows stores all of this as Unicode, so when you call the "A" versions it's converting the Unicode to ANSI and then we're having to convert it back.

You can't use `ExactSpelling` unless the name actually matches. The DLL entry points are actually "QueryDosDeviceA" and "QueryDosDeviceW" for this API. The Windows headers have "QueryDosDevice" defined as one of those two depending on whether or not you've defined `UNICODE`. .NET probes for `A` or `W` names when it can't find the name you've given to make the process feel like writing C/C++ (there is no export called "QueryDosDevice"). We're preferring to be explicit in CoreFX to avoid accidentally using ANSI when we should almost always be using Unicode. (Like 99% of the time.)

Reply:
Interesting to find QueryDosDeviceW returns a double null terminated string unlike QueryDosDevice
Anyway, handled that ok I think.

Reply:
@PRIMETSS I suspect double 0 is because Unicode uses two bytes per character so it's actually one character which is 0 producing 2 bytes (although just a guess)

Reply:
Yes, they both return a double null. It's a common Windows pattern for lists of strings. In ANSI it's `0x00 0x00` in bytes, in Unicode it's `0x00 0x00 0x00 0x00`.